android/src/main/kotlin/atomic/financial/atomic_transact_flutter/AtomicTransactFlutterPlugin.kt

```diff
@@ -35,31 +34,49 @@ class AtomicTransactFlutterPlugin: FlutterPlugin, MethodCallHandler, ActivityAwa
       val configuration = call.argument<Map<String, Any>>("configuration")
       val publicToken = configuration?.get("publicToken") as String
       val language = configuration?.get("language") as String
-      val product = configuration?.get("product") as String
+      val tasks = configuration?.get("tasks") as? List<Map<String, Any>>
+      val product = configuration?.get("product") as? String
       val additionalProduct = configuration?.get("additionalProduct") as? String
+      val distribution = configuration?.get("distribution") as? Map<String, Any>
       val handoff = configuration?.get("handoff") as? List<String>
       val linkedAccount = configuration?.get("linkedAccount") as? String
       val metadata = configuration?.get("metadata") as? String
       val theme = configuration?.get("theme") as? Map<String, Any>
-      val distribution = configuration?.get("distribution") as? Map<String, Any>
       val deeplink = configuration?.get("deeplink") as? Map<String, Any>
       val search = configuration?.get("search") as? Map<String, Any>
       val experiments = configuration?.get("experiments") as? Map<String, Any>
 
-      val config = Config(
-              publicToken = publicToken,
-              product = Config.Product.valueOf(product.toUpperCase()),
-              additionalProduct = if (additionalProduct != null) Config.Product.valueOf(additionalProduct.toUpperCase()) else null,
-              linkedAccount = linkedAccount,
-              handoff = configHandoffFromList(handoff),
-              language = Config.Language.valueOf(language),
-              metadata = metadata,
-              theme = configThemeFromMap(theme),
-              distribution = configDistributionFromMap(distribution),
-              deeplink = configDeeplinkFromMap(deeplink),
-              experiments = configExperimentsFromMap(experiments),
-              search = configSearchFromMap(search)
-      )
+      val config : Config
+
+      if (product != null) {
+        config = Config(
```

Comment:
@jorgefspereira since the config is basically the same except for each instance could we simplify this a bit? The only difference is `tasks` and `product`. Would it may sense to have the if statement either inject `product` or `tasks` into the config instead of having a repeat of all of the items?

Reply:
@brennens You're right, it should be simpler. But for that the Config object of the Atomic Android SDK should implement the Builder Pattern to allow conditional injection of the parameters.